builder/dockerfile/dispatchers.go

```diff
@@ -651,6 +638,17 @@ func arg(b *Builder, args []string, attributes map[string]bool, original string)
 	return b.commit("", b.runConfig.Cmd, fmt.Sprintf("ARG %s", arg))
 }
 
+// SHELL
+//
+// Set the shell used for RUN commands.
+func shell(b *Builder, args []string, attributes map[string]bool, original string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("SHELL requires exactly one argument definition")
+	}
+	b.runConfig.Shell = strings.Split(args[0], " ")
```

Comment:
I think you'd be better off letting the dockerfile parser split these so that it can deal with quotes, spaces, etc...
If there isn't a Node parser that just calls `parseWords()` then you may need to create one - but that should be trivial.